test/operations_test.cpp

```diff
@@ -49,6 +49,10 @@ using std::endl;
 #include <ctime>
 #include <cstdlib> // for system(), getenv(), etc.
 
+#if defined(BOOST_POSIX_API) && defined(__linux__)
+# include <linux/version.h>
```

Comment:
Is this necessary? Could you run the tests unconditionally and simply skip them if you get `ENOSYS`?